### PR TITLE
Fix Places where AccountId Uses Default

### PR DESCRIPTION
**Pull request by shawntabrizi**
Fixes: https://github.com/paritytech/substrate/issues/10554

### CODE REVIEW

**Review by bkchr — APPROVED** [review]
Why didn't this failed in ci? 🤔

File: frame/system/src/lib.rs [hunk 1]
```diff
@@ -833,15 +833,17 @@ impl<O: Into<Result<RawOrigin<AccountId>, O>> + From<RawOrigin<AccountId>>, Acco
...
+		O::from(RawOrigin::Signed(zero_account_id))
 	}
 }
 
 pub struct EnsureSignedBy<Who, AccountId>(sp_std::marker::PhantomData<(Who, AccountId)>);
 impl<
 		O: Into<Result<RawOrigin<AccountId>, O>> + From<RawOrigin<AccountId>>,
 		Who: SortedMembers<AccountId>,
-		AccountId: PartialEq + Clone + Ord + Default,
+		AccountId: PartialEq + Clone + Ord,
```
Comment (bkchr) [inline frame/system/src/lib.rs:846]:
```suggestion
		AccountId: PartialEq + Clone + Ord + Decode,
```

**Comment by bkchr** [comment]
> Why didn't this failed in ci? 🤔

A probably because it isn't used anywhere?

**Review by KiChjang — APPROVED** [review]

**Comment by xlc** [comment]
Some additional tests?

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**